packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -1197,6 +1320,63 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
             // This highlights the left side if the row is loading.
             this.setState({ selectedRow: id });
         }
+
+        if (this.state.ctxMenu) {
+            this.setState({ selectedRows: [id] });
+            signalManager().fireRowSelectionsChanged({
+                traceId: this.props.traceId,
+                outputDescriptor: this.props.outputDescriptor,
+                rowIds: [id]
+            });
+        }
+    };
+
+    public onMultipleRowClick = (id: number, isShiftClicked?: boolean): void => {
+        const tree = listToTree(this.state.timegraphTree, this.state.columns);
+        const rowIndex = getIndexOfNode(id, tree, this.state.collapsedNodes);
+        this.chartLayer.selectAndReveal(rowIndex);
+
+        const rows = this.state.selectedRows ? this.state.selectedRows : [];
+        const lastSelectedRow = this.state.selectedRows?.at(this.state.selectedRows.length - 1);
+        let lastSelectedRowIndex = undefined;
+        if (lastSelectedRow) {
+            lastSelectedRowIndex = getIndexOfNode(lastSelectedRow, tree, this.state.collapsedNodes);
+        }
+        if (isShiftClicked && lastSelectedRowIndex) {
+            let startIndex = 0;
+            let endIndex = 0;
+            if (lastSelectedRowIndex < rowIndex) {
+                startIndex = lastSelectedRowIndex;
+                endIndex = rowIndex;
+            } else {
+                startIndex = rowIndex;
+                endIndex = lastSelectedRowIndex;
+            }
+
+            const nodeIds = getAllExpandedNodeIds(tree, this.state.collapsedNodes);
+            if (startIndex < 0 || startIndex >= nodeIds.length || endIndex >= nodeIds.length) {
+                return;
+            }
+            for (let i = startIndex; i <= endIndex; i++) {
+                const nodeId = nodeIds.at(i);
+                if (nodeId) {
+                    rows.push(nodeId);
+                }
+            }
+        } else {
+            if (rows.includes(id)) {
+                const index = rows.indexOf(id);
+                rows.splice(index, 1);
+            } else {
+                rows?.push(id);
+            }
+        }
+        this.setState({ selectedRow: undefined, selectedRows: [...rows] });
```

Comment:
I primiarily did this for styling of the table row.. I wanted all multi selection of rows to be the same color - and if i dont set selectedRow to be undefined, the className that gets attached to the row would be selected as that was my 2nd check for what className should be used.

But on second thought, I dont think multi-select needs to be a different background. It can just re-use the same background as the normal selection. WDYT?